torch/csrc/api/src/nn/modules/rnn.cpp

```diff
@@ -919,26 +898,39 @@ RNNCellImpl::RNNCellImpl(const RNNCellOptions& options_)
       options(options_) {}
 
 Tensor RNNCellImpl::forward(const Tensor& input, Tensor hx) {
-  this->check_forward_input(input);
+  this->check_forward_input(input, "input");
+  this->check_forward_input(hx, "hidden");
```

Comment:
Do we need to check whether `hx.defined()` before this check?

Reply:
Thank you for your comment.

I am sorry for late reply. I think there is no need to check hx.define(), because all undefined tensors will share an impl, and the dim of this impl will be 1 by default, so whether it is defined or not, the requirement [1, 2] will be satisfied

Reply:
One thing to add is that the lack of the hx.define() check will indeed lead to some ambiguity